Dockerfile

```diff
@@ -0,0 +1,36 @@
+## Build
+FROM golang:1.13-buster AS build
+
+COPY bigtable /src/bigtable
+COPY storage /src/storage
+
+WORKDIR /src/bigtable/
+RUN go mod download
+RUN go build -o /cbtemulator ./cmd/cbtemulator
+
+WORKDIR /src/storage/
+RUN go mod download
+RUN go build -o /gcsemulator ./cmd/gcsemulator
+
+## Deploy
+FROM gcr.io/distroless/base-debian10 AS cbtemulator
+
+WORKDIR /
+
+COPY --from=build /cbtemulator /cbtemulator
+
+EXPOSE 9000
+
+ENTRYPOINT ["/cbtemulator", "-port", "9000", "-host", "0.0.0.0", "-dir", "var/bigtable"]
+
+
+## Deploy
+FROM gcr.io/distroless/base-debian10 AS gcsemulator
```

Comment:
Oh interesting. I didn't know you could produce two containers from a single `Dockerfile`. I thought the different stages were just discarded after all steps in the file were run.

Reply:
normally all stages but the last one are discarded.

if you check the script file that scott added, when building we specify a target, so it build the image out of that target, that's how we control the output